.github/workflows/main.yml

```diff
@@ -0,0 +1,47 @@
+# This is a basic workflow to help you get started with Actions
+
+name: tweet_scheduler
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [ main ]
+  #schedule:
+  #  - cron: '30 10,18 * * *'    
+
+jobs:
+  hadleyfreebot-post:
+    runs-on: macOS-latest
+    env:
+      TWITTER_CONSUMER_API_KEY_HADLEYBOT: ${{ secrets.TWITTER_CONSUMER_API_KEY_HADLEYBOT }}
+      TWITTER_CONSUMER_API_SECRET_HADLEYBOT: ${{ secrets.TWITTER_CONSUMER_API_SECRET_HADLEYBOT }}
+      TWITTER_ACCESS_TOKEN_HADLEYBOT: ${{ secrets.TWITTER_ACCESS_TOKEN_HADLEYBOT }}
+      TWITTER_ACCESS_TOKEN_SECRET_HADLEYBOT: ${{ secrets.TWITTER_ACCESS_TOKEN_SECRET_HADLEYBOT }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/setup-r@master
+
+  #Install R packages
+      - name: Install packages
+        run: |
+          R -e 'install.packages("here", dependencies = TRUE)'
+          R -e 'install.packages("readr", dependencies = TRUE)'
+          R -e 'install.packages("glue", dependencies = TRUE)'
+          R -e 'install.packages("rtweet", dependencies = TRUE)'
+  
+  #Run R script
+      - name: Create and post tweet
+        run: Rscript hadley_bot_script.R
+
+  #Commit updated files to repo
+      - name: commit files
+        run: |
+          git config --local user.name actions-user
+          git config --local user.email "actions@github.com"
+          git add Code/Data/*
+          git commit -am "GA updates rows tweeted dataset"
+          git push origin main
```

Comment:
currently this GH Action is set to run when main branch is updated. If you push here you will create a recursion loop where it keeps running.